chia/rpc/wallet_rpc_api.py

```diff
@@ -153,31 +153,19 @@ def get_routes(self) -> Dict[str, Callable]:
             "/pw_status": self.pw_status,
         }
 
-    async def _state_changed(self, *args) -> List[WsRpcMessage]:
+    async def _state_changed(self, change: str, change_data: Dict[str, Any]) -> List[WsRpcMessage]:
         """
         Called by the WalletNode or WalletStateManager when something has changed in the wallet. This
         gives us an opportunity to send notifications to all connected clients via WebSocket.
         """
         payloads = []
-        if args[0] is not None and args[0] == "sync_changed":
+        if change in {"sync_changed", "coin_added"}:
```

Comment:
I think that handling `"coin_added"` here maybe misses it having the `"state"` key in the `change_data` dict?